README.md

```diff
@@ -42,87 +42,76 @@ Find a list of all the available options for your DogStatsD Client in the [DogSt
 
 2. You have to make sure you are either:
 
-  * using singletons instances of the DogStatsD client and not allocating one each time you need one, letting the buffering mechanism flush metrics, it's still a bad solution if the process later forks (see related section below). Or,
-  * properly closing your DogStatsD client instance when it is not needed anymore using the method `Datadog::Statsd#close` to release the resources used by the instance and to close the socket
+  * Using a singleton instance of the DogStatsD client instead of creating a new instance whenever you need one; this will let the buffering mechanism flush metrics regularly
+  * Or properly disposing of the DogStatsD client instance when it is not needed anymore using the method `Datadog::Statsd#close`
 
 If you have issues with the companion thread or the buffering mode, you can instantiate a client that behaves exactly as in v4.x (i.e. no companion thread and flush on every metric submission):
 
 ```ruby
-# Import the library
-require 'datadog/statsd'
-
 # Create a DogStatsD client instance using UDP
 statsd = Datadog::Statsd.new('localhost', 8125, single_thread: true, buffer_max_pool_size: 1)
-...
-# to close the instance is not necessary in this case since metrics are flushed on submission
-# but it is still a good practice and it explicitely closes the socket
+# ...
 statsd.close()
 ```
 
 or
 
 ```ruby
-# Import the library
-require 'datadog/statsd'
-
 # Create a DogStatsD client instance using UDS
 statsd = Datadog::Statsd.new(socket_path: '/path/to/socket/file', single_thread: true, buffer_max_pool_size: 1)
-...
-# to close the instance is not necessary in this case since metrics are flushed on submission
-# but it is still a good practice and it explicitely closes the socket
+# ...
 statsd.close()
 ```
 
 ### v5.x Common Pitfalls
 
-Version v5.x of `dogstatsd-ruby` is using a companion thread for preemptive flushing, it brings better performances for application having a high-throughput of statsd metrics, but it comes with new pitfalls:
+Version v5.x of `dogstatsd-ruby` is using a companion thread for flushing. This provides better performance, but you need to consider the following pitfalls:
 
-    * Applications forking after having created the dogstatsd instance: forking a process can't duplicate the existing threads, meaning that one of the processes won't have a companion thread to flush the metrics and will lead to missing metrics.
-    * Applications creating a lot of different instances of the client without closing them: it is important to close the instance to free the thread and the socket it is using or it will lead to thread leaks.
+1. Applications that use `fork` after having created the dogstatsd instance: the child process will need a new dogstatsd instance, as otherwise it will be unable to send metrics.
```

Comment:
The reasoning why this is the case is still very useful I think

Reply:
As a user of the library, how relevant is it for me to learn about this in the "common pitfalls" section? This seems to go really deep into library development details, which I'm not entirely sure belong in this section of the README.

Reply:
It's a really big and likely pitfall. Most Ruby deployments use `puma` or `unicorn` that `fork()` the processes so having that note in a different part of the readme I think would cause a lot of heartburn for the users that won't be obvious until long after their code is deployed.

Reply:
Not true anymore since 5.3.0, I have to remove that line.